### PR TITLE
[Silabs] Update Light-switch Data Model

### PR DESCRIPTION
**Pull request by jepenven-silabs**
Create a custom data model for silabs light-switch app to leverage WIP ICD features.

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

File: examples/light-switch-app/silabs/data_model/light-switch-app.matter [hunk 1]
```diff
@@ -0,0 +1,2704 @@
...
+  server cluster Descriptor {
+    callback attribute deviceTypeList;
+    callback attribute serverList;
+    callback attribute clientList;
+    callback attribute partsList;
+    ram      attribute featureMap default = 0;
+    callback attribute clusterRevision default = 1;
+  }
+
+  server cluster Binding {
```
Comment (bzbarsky-apple) [inline examples/light-switch-app/silabs/data_model/light-switch-app.matter:2325]: Why does this have a Binding server on EP0?

File: examples/light-switch-app/silabs/data_model/light-switch-app.matter [hunk 2]
```diff
@@ -0,0 +1,2704 @@
...
+  }
+
+  server cluster LocalizationConfiguration {
+    ram      attribute activeLocale;
+    callback attribute supportedLocales;
+    ram      attribute featureMap default = 0;
+    ram      attribute clusterRevision default = 1;
+  }
+
+  server cluster TimeFormatLocalization {
```
Comment (bzbarsky-apple) [inline examples/light-switch-app/silabs/data_model/light-switch-app.matter:2390]: Does this switch in fact implement TimeFormatLocalization?

File: examples/light-switch-app/silabs/data_model/light-switch-app.matter [hunk 3]
```diff
@@ -0,0 +1,2704 @@
...
+    ram      attribute connectMaxTimeSeconds;
+    ram      attribute interfaceEnabled;
+    ram      attribute lastNetworkingStatus;
+    ram      attribute lastNetworkID;
+    ram      attribute lastConnectErrorValue;
+    ram      attribute featureMap default = 2;
+    ram      attribute clusterRevision default = 1;
+  }
+
+  server cluster DiagnosticLogs {
```
Comment (bzbarsky-apple) [inline examples/light-switch-app/silabs/data_model/light-switch-app.matter:2421]: I really doubt this implements DiagnosticLogs.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/light-switch-app/silabs/data_model/light-switch-app.matter (hunk 1 above)
Comment (bzbarsky-apple) [reply examples/light-switch-app/silabs/data_model/light-switch-app.matter:2325]: I think we need to stop having broken and non-spec-conformant examples.

We should fix it here (whether in this PR in a followup is fine with me, as long as it happens).  We should fix the broken thing this got copy/pasted from (separate PR).

**Comment by jepenven-silabs** [comment]
close as not relevant for the short term

**Pull request closed without merging**